### ISSUE TITLE
Update wasmer?

### ISSUE THREAD

**Issue by gaborcsardi** [dirkschumacher/wasmrrr#22]
I cannot build it now on macOS with the latest (`rustup update`) tools:

```
error[E0425]: cannot find function `get_fault_info` in this scope
   --> /Users/gaborcsardi/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmer-runtime-core-0.17.1/src/fault.rs:289:21
    |
289 |         let fault = get_fault_info(siginfo as _, ucontext);
    |                     ^^^^^^^^^^^^^^ not found in this scope
```

updating the wasmer version to 0.17.1 does not help, either.

Do you think it is possible to move to the new wasmer crate?

**Comment by dirkschumacher** [comment]
Yeah definitely possible. I haven't touched it since I got a Mac M1 (#20), because I could not compile it. I also have not followed the development of wasmer (and wasmtime) in recent months and things will likely break I guess.

**Comment by gaborcsardi** [comment]
I don't think the issue is M1 related, but I'll try it on Intel as well. I think `wasmer-runtime-core` is simply not supported any more, and wasmer is the way to go now. At least this is my high level understanding.

**Comment by syrusakbary** [comment]
Latest wasmer should support and compile on macOS M1 (I'm writing here from a M1 Max!), let me know if we can help on anything there :)

**Comment by dirkschumacher** [comment]
Ok, will try to make a PR 🤞

**Comment by dirkschumacher** [comment]
The current implementation relies heavily on the API exposed in [wasmer-c-api](https://github.com/wasmerio/wasmer-c-api). Good examples are now here: https://docs.wasmer.io/integrations/examples

I think it will make sense to rewrite the C++ part.

**Comment by dirkschumacher** [comment]
I think it make sense to completely rewrite it and move it very closely to the Wasmer API, similar to the other language bindings. With classes/functions for concepts like `Engine`, `Store`, `Module`, etc.
